api/nodes/permissions.py

```diff
@@ -15,6 +17,34 @@ def has_object_permission(self, request, view, obj):
             return obj.can_edit(auth)
 
 
+class AdminOrPublic(permissions.BasePermission):
+
+    def has_object_permission(self, request, view, obj):
+        assert isinstance(obj, (Node, User)), 'obj must be a Node or User, got {}'.format(obj)
+        auth = get_user_auth(request)
+        node = Node.load(request.parser_context['kwargs']['node_id'])
+        if request.method in permissions.SAFE_METHODS:
+            return node.is_public or node.can_view(auth)
+        else:
+            return node.has_permission(auth.user, osf_permissions.ADMIN)
+
+
+class ContributorDetailPermissions(permissions.BasePermission):
+    """Permissions for contributor detail page."""
+
+    def has_object_permission(self, request, view, obj):
+        assert isinstance(obj, (Node, User)), 'obj must be User or Node, got {}'.format(obj)
+        auth = get_user_auth(request)
+        node = Node.load(request.parser_context['kwargs']['node_id'])
+        user = User.load(request.parser_context['kwargs']['user_id'])
+        if request.method in permissions.SAFE_METHODS:
+            return node.is_public or node.can_view(auth)
+        elif request.method == 'DELETE':
+            return node.has_permission(auth.user, osf_permissions.ADMIN) or auth.user == user
```

Comment:
What is the use case for the `auth.user == user` branch? It seems as though that would allow deletion when the logged-in user was the same as the user ID specified in the URL. That seems a rather... fragile... test for permission.

We have other permissions checks to keep this from being too sketchy in practice- but is that branch needed at all?